src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,544 @@
+                        for (int i = 0; i < 5; i++)
+                        {
+                            letter[i] = NokiaCharacters.Ascii[c - 0x20][i];
+                        }
+
+                        letter[5] = 0x00;
+                        SpiWrite(true, letter.ToArray());
+                        _position += 5;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Write text and set cursor position to next line
+        /// </summary>
+        /// <param name="text">The text to write</param>
+        public void WriteLine(string text)
+        {
+            Write(text);
+            // calculate the position
+            int y = _position / Size.Width + 1;
+            y = y > Size.Height ? Size.Height : y;
+            SetCursorPosition(0, y);
+        }
+
+        /// <summary>
+        /// Write a raw byte stream to the display.
+        /// Used if character translation already took place.
+        /// </summary>
+        /// <param name="text">Text to print</param>
+        public void Write(ReadOnlySpan<byte> text)
+        {
+            var length = text.Length > ScreenBufferByteSize - _position ? ScreenBufferByteSize - _position : text.Length;
+            text.Slice(0, length).CopyTo(_byteMap.AsSpan(_position));
+            Draw();
+        }
+
+        /// <summary>
+        /// Moves the cursor to an explicit column and row position.
+        /// </summary>
+        /// <param name="left">The column position from left to right starting with 0 to 83.</param>
+        /// <param name="top">The row position from the top starting with 0 to 5.</param>
+        /// <exception cref="ArgumentOutOfRangeException">The given position is not inside the display.</exception>
+        public void SetCursorPosition(int left, int top)
+        {
+            if ((left < 0 || left > Size.Width) || (top < 0 || top > Size.Height))
+            {
+                throw new ArgumentOutOfRangeException($"The given position is not inside the display. it's 6 raws and 84 columns");
+            }
+
+            _position = left + top * Size.Width;
+            SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff), (byte)((byte)(SetAddress.XAddress) | left), (byte)((byte)(SetAddress.YAddress) | top) });
+        }
+
+        #endregion
+
+        #region Drawing points, lines and rectangles
+
+        /// <summary>
+        /// Draw a point
+        /// </summary>
+        /// <param name="x">The X coordinate.</param>
+        /// <param name="y">The Y coordinate.</param>
+        /// <param name="isOn">True if the point has pixels on, false for off.</param>
+        /// <returns>True if success</returns>
+        public bool DrawPoint(int x, int y, bool isOn)
+        {
+            if (x < 0 || x >= 84 || y < 0 || y >= 48)
+            {
+                return false;
+            }
+
+            int index = ((x % 84) + (int)(y * 0.125) * 84);
```

Comment:
any reason you picked to do `(int)(y * 0.125)` rather than `y / 8`?

Reply:
yes, if I use /8, I will not get a proper rounding.

Reply:
Can you give specific example? I've never had to use floating points in routines like this. (I'm ok with this as is, just curious)

Reply:
ho, I think this is what I originally wanted to do: int index = (int)((x % 84) + (y * 0.125) * 84);
and that's different. Need to test again that it's the expected behavior with the actual code, in that case, yes, the /8 will work perfectly!